core/src/main/java/org/apache/iceberg/InternalData.java

```diff
@@ -163,6 +180,11 @@ public interface ReadBuilder {
     /** Set a custom class for in-memory objects at the given field ID. */
     ReadBuilder setCustomType(int fieldId, Class<? extends StructLike> structClass);
 
+    /** Set the classloader used for custom types. */
+    default ReadBuilder classLoader(ClassLoader classLoader) {
+      throw new UnsupportedOperationException("Classloader not supported");
```

Comment:
I don't think that this is needed. I originally added it in the InternalData commit, but because we are passing the classes themselves (rather than loading them dynamically by name) they are already loaded.

Reply:
I couldn't find why this was necessary since the builder itself should have the same classloader.  I was thinking there must be some usage elsewhere but I don't see anything.  I'll remove this.